src/main/java/org/broadinstitute/hellbender/tools/exome/orientationbiasvariantfilter/OrientationBiasFilterer.java

```diff
@@ -0,0 +1,361 @@
+ * Perform operations for the orientation bias filter.
+ */
+public class OrientationBiasFilterer {
+
+    private static final Logger logger = LogManager.getLogger(OrientationBiasFilterer.class);
+
+    //TODO: Promote or replace with VCFConstants call
+    public static final String ALT_F2R1 = "ALT_F2R1";
+    public static final String ALT_F1R2 = "ALT_F1R2";
+
+    public static final Double PRE_ADAPTER_METRIC_NOT_ARTIFACT_SCORE = 100.0;
+    public static final double BIASP = 0.96;
+
+    /** Adds the annotations that can be created for individual variants.  The annotations placed on the genotypes here will be used by the filter.
+     * @param variant the variant to prepare for the orientation bias filter. Never {@code null}
+     * @param relevantArtifactModesWithoutComplement the SNV artifact modes that we want to filter against.  Do not include the complement.  Never {@code null}
+     * @param preAdapterQScoreMap mapping from Artifact mode to the preAdapterQ score.  Never {@code null}
+     * @return updated VariantContext with new genotypes already populated.
+     */
+    public static VariantContext annotateVariantContextWithPreprocessingValues(final VariantContext variant, final SortedSet<ArtifactMode> relevantArtifactModesWithoutComplement, final Map<ArtifactMode, Double> preAdapterQScoreMap) {
+
+        Utils.nonNull(variant);
+        Utils.nonNull(relevantArtifactModesWithoutComplement);
+        Utils.nonNull(preAdapterQScoreMap);
+
+        final List<ArtifactMode> relevantArtifactModesComplement = OrientationBiasUtils.createReverseComplementArtifactModes(relevantArtifactModesWithoutComplement);
+
+        VariantContextBuilder variantBuilder = new VariantContextBuilder(variant);
+        GenotypesContext genotypesContext = variant.getGenotypes();
+
+        final List<Genotype> newGenotypes = new ArrayList<>();
+        for (Genotype genotype: genotypesContext.iterateInSampleNameOrder()) {
+            final GenotypeBuilder genotypeBuilder = new GenotypeBuilder(genotype);
+            genotypeBuilder.attribute(OrientationBiasFilterConstants.IS_ORIENTATION_BIAS_ARTIFACT_MODE, String.valueOf(false));
+            genotypeBuilder.attribute(OrientationBiasFilterConstants.IS_ORIENTATION_BIAS_RC_ARTIFACT_MODE, String.valueOf(false));
+
+            final List<Allele> alleles = genotype.getAlleles();
+            if (genotype.getPloidy() != 2) {
+                throw new UserException.BadInput("This tool will not run with non-diploid organisms.  Saw GT: " + genotype.getGenotypeString());
+            }
+
+            // Get the reference allele as a String
+            final List<String> refAlleles = alleles.stream().filter(a -> a.isReference()).map(a -> a.getBaseString()).collect(Collectors.toList());
+            if (!((refAlleles.size() != 1) || (refAlleles.get(0).length() != 1))) {
+                final Character refAllele = (char) refAlleles.get(0).getBytes()[0];
+                for (int i = 1; i < alleles.size(); i ++ ) {
```

Comment:
I need the counter (``i``) for the ``totalAltAlelleCount`` unless I am missing something...